addons/l10n_ro_edi/models/account_edi_xml_ubl_ciusro.py

```diff
@@ -0,0 +1,105 @@
+        vals = super()._get_partner_address_vals(partner)
+
+        if partner.state_id:
+            vals["country_subentity"] = partner.country_code + '-' + partner.state_id.code
+
+        return vals
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+
+        if invoice.currency_id.name != 'RON':
+            ron_currency = self.env.ref("base.RON")
+            vals_list.append({
+                'currency': ron_currency,
+                'currency_dp': ron_currency.decimal_places,
+                'tax_amount': taxes_vals['tax_amount'],
+            })
+
+        return vals_list
+
+    def _get_partner_party_tax_scheme_vals_list(self, partner, role):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_partner_party_tax_scheme_vals_list(partner, role)
+
+        if not partner.vat and partner.company_registry:
+            company_registry = partner.company_registry
+
+            return [{
+                'company_id': company_registry,
+                'TaxScheme_vals': {},
+                'tax_scheme_id': 'VAT',
+            }]
+
+        return vals_list
+
+    def _get_partner_party_legal_entity_vals_list(self, partner):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_partner_party_legal_entity_vals_list(partner)
+
+        for vals in vals_list:
+            if not partner.vat and partner.company_registry:
+                vals['company_id'] = partner.company_registry
+
+        return vals_list
+
+    def _export_invoice_vals(self, invoice):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals = super()._export_invoice_vals(invoice)
+
+        vals['vals'].update({
+            'customization_id': 'urn:cen.eu:en16931:2017#compliant#urn:efactura.mfinante.ro:CIUS-RO:1.0.1',
+            'tax_currency_code': 'RON',
+        })
+
+        return vals
+
+    def _export_invoice_constraints(self, invoice, vals):
+        # EXTENDS account.edi.xml.ubl_bis3
+        constraints = super()._export_invoice_constraints(invoice, vals)
+
+        for partner_type in ('supplier', 'customer'):
+            partner = vals[partner_type]
+
+            constraints.update({
+                f"ciusro_{partner_type}_city_required": self._check_required_fields(partner, 'city'),
+                f"ciusro_{partner_type}_street_required": self._check_required_fields(partner, 'street'),
+                f"ciusro_{partner_type}_state_id_required": self._check_required_fields(partner, 'state_id'),
+            })
+
+            if not partner.vat and not partner.company_registry:
+                constraints[f"ciusro_{partner_type}_tax_identifier_required"] = _(
+                    "The following partner doesn't have a VAT nor Company ID: %s. "
+                    "At least one of them is required. "
+                ) % partner.name
```

Comment:
So we no longer verify that the company_registry starts with the country_code ? I thought it was required ?
Same for the VAT.

Reply:
@JulienVR  sorry about this :laughing: at first I thought this was actually not needed but I rechecked again and turns out the rule actually exist. I added it on the newest commit :)

Reply:
@yosuanicolaus Don't forget to squash it when the pr is ready :smile: And for next time instead of making a new commit for reviewer to check you can check how git amend works :smile: